### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by ryao**
### Motivation and Context
CodeQL and Coverity pointed out some cleanup opportunities.

### Description
See the individual commit messages.

### How Has This Been Tested?
The buildbot can test it.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [x] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Library ABI change (libzfs, libzfs\_core, libnvpair, libuutil and libzfsbootenv)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the OpenZFS [code style requirements](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**contributing** document](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/openzfs/zfs/tree/master/tests) to cover my changes.
- [ ] I have run the ZFS Test Suite with this change applied.
- [x] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

**Review by szubersk — APPROVED** [review]

File: cmd/zdb/zdb.c [hunk 1]
```diff
@@ -4360,7 +4360,7 @@ dump_l2arc_log_blkptr(l2arc_log_blkptr_t lbps)
 }
 
 static void
-dump_l2arc_log_blocks(int fd, l2arc_dev_hdr_phys_t l2dhdr,
+dump_l2arc_log_blocks(int fd, l2arc_dev_hdr_phys_t *l2dhdr,
```
Comment (amotin) [inline cmd/zdb/zdb.c:4363]: I'd add const here and in dump_l2arc_log_entries() just for clarity.

**Review by amotin — APPROVED** [review]

**Comment by ryao** [comment]
The repush fixes checkstyle complaints.

File: cmd/zdb/zdb.c (hunk 1 above)
Comment (ryao) [reply cmd/zdb/zdb.c]: Good idea. Also, nice catch on `dump_l2arc_log_blkptr()`. Coverity failed to report that one. Changing it to use a constant pointer means we can cleanup the weird `(&lbps)->lbp_prop`.

I have pushed a revision incorporating these changes.

**Comment by ryao** [comment]
@szubersk The zdb patch has been repushed with changes suggested by @amotin. Just a FYI, since you approved an earlier version of it.

**Review by amotin — APPROVED** [review]

**Comment by behlendorf** [comment]
You've probably already seen this, but this change appears to have introduced the following two new failures:

```
 Tests with results other than PASS that are unexpected:
    FAIL cli_root/zpool_labelclear/zpool_labelclear_removed (expected PASS)
    FAIL cli_root/zpool_labelclear/zpool_labelclear_valid (expected PASS)
```

**Review by freqlabs — APPROVED** [review]

**Pull request closed without merging**